### PR TITLE
[GPU] Add 2d load/store ops validation

### PR DESCRIPTION
**Pull request by kurapov-peter**
This adds the necessary validation for HW parameters of the memory operations and diagnostic tests.

Depends on #389

### CODE REVIEW

File: lib/gc/Dialect/LLVMIR/IR/XeVMDialect.cpp [hunk 1]
```diff
@@ -9,13 +9,270 @@
...
+    return op->emitOpError(
+        "4th operand (base pitch) should be >= 2nd operand (base width)");
+
+  if (op.getElemSizeInBits() != 8 && op.getElemSizeInBits() != 16 &&
+      op.getElemSizeInBits() != 32)
+    return op->emitOpError("expecting 'elem_size_in_bits' to be 8, 16, or 32");
+
+  uint32_t tileHeight = op.getTileHeight();
+  if (tileHeight != 1 && tileHeight != 2 && tileHeight != 4 &&
+      tileHeight != 8 && tileHeight != 16 && tileHeight != 32)
```
Comment (AndreyPavlenko) [inline lib/gc/Dialect/LLVMIR/IR/XeVMDialect.cpp:41]:
```suggestion
  if (tileHeight > 32 || tileHeight & (tileHeight - 1))
```

File: lib/gc/Dialect/LLVMIR/IR/XeVMDialect.cpp [hunk 2]
```diff
@@ -9,13 +9,270 @@
...
+    return op->emitOpError("expecting tile_height to be 1, 2, 4, 8, 16, or 32");
+
+  uint32_t vBlocks = op.getVBlocks();
+  if (vBlocks != 1 && vBlocks != 2 && vBlocks != 4 && vBlocks != 8)
```
Comment (AndreyPavlenko) [inline lib/gc/Dialect/LLVMIR/IR/XeVMDialect.cpp:45]:
```suggestion
if (vBlocks > 8 || vBlocks & (vBlocks - 1))
```

File: lib/gc/Dialect/LLVMIR/IR/XeVMDialect.cpp [hunk 3]
```diff
@@ -9,13 +9,270 @@
...
+    return op->emitOpError("expecting v_blocks to be 1, 2, 4, or 8");
+
+  return success();
+}
+
+LogicalResult verify2DBlockLoadHWRestriction(BlockLoad2dOp op) {
+  VectorType resTy = op.getRes().getType();
+  unsigned resElemTySize = resTy.getElementType().getIntOrFloatBitWidth();
```
Comment (AndreyPavlenko) [inline lib/gc/Dialect/LLVMIR/IR/XeVMDialect.cpp:53]: Is it possible non-int or float here?

**Review by AndreyPavlenko — APPROVED** [review]

File: lib/gc/Dialect/LLVMIR/IR/XeVMDialect.cpp (hunk 3 above)
Comment (kurapov-peter) [reply lib/gc/Dialect/LLVMIR/IR/XeVMDialect.cpp:56]: Nope, nice catch. I added a check.

File: lib/gc/Dialect/LLVMIR/IR/XeVMDialect.cpp (hunk 2 above)
Comment (kurapov-peter) [reply lib/gc/Dialect/LLVMIR/IR/XeVMDialect.cpp]: I've replaced those with a similar check though I think overall it gets less readable. All others seem to get messier if changed so I kept them intact.

**Review by dchigarev — APPROVED** [review]

**Pull request merged**